### PR TITLE
1553 read only rhs toast on checklist

### LINKED ISSUE #1553: Pop up read only RHS toast when user attempts to check off a checklist item

### PR DESCRIPTION
**Pull request by gbochora**
## Summary
This PR adds toast when a channel member with read-only access to the run tries to change the checklist state.

https://user-images.githubusercontent.com/4368372/200803827-ff705ebd-d090-4ecd-bf88-be2ee675e828.mp4

## Ticket Link
Fixes #1553

## Checklist

~~- [ ] Telemetry updated~~
~~- [ ] Gated by experimental feature flag~~
~~- [ ] Unit tests updated~~

### CODE REVIEW

**Review by crspeller — APPROVED** [review]
Nice :)

**Comment by trilopin** [comment]
it's triggering the toast when the run is finished, shouldn't be completely disabled in that case?

File: webapp/src/components/checklist_item/inputs.tsx [hunk 1]
```diff
@@ -30,6 +31,10 @@ export const CheckBoxButton = (props: CheckBoxButtonProps) => {
     //     Additionally, we prevent the user from clicking multiple times
     //     and leaving the item in an unknown state
     const handleOnChange = async () => {
+        if (props.readOnly) {
+            props.onChange(isChecked ? ChecklistItemState.Closed : ChecklistItemState.Open);
```
Comment (trilopin) [inline webapp/src/components/checklist_item/inputs.tsx:35]: do we have to invoke this onChange without changing it at all? can we skip this?

**Review by trilopin — APPROVED** [review]
Cool

I left a couple of non-blocking comments

File: webapp/src/components/checklist_item/inputs.tsx (hunk 1 above)
Comment (gbochora) [reply webapp/src/components/checklist_item/inputs.tsx]: Yep, I've passed `onViewerModeInteract` to the `CheckBoxButton` component, so there is no need to handle click in the parent component.

**Pull request merged**